api/src/main/java/org/openmrs/ConceptReferenceRange.java

```diff
@@ -0,0 +1,277 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import org.codehaus.jackson.annotate.JsonIgnore;
+import org.hibernate.envers.Audited;
+import org.hibernate.search.annotations.DocumentId;
+import org.hibernate.search.annotations.Field;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import java.util.Date;
+
+/**
+ * A concept reference range is typically a range of a concept for certain factor(s) e.g. age, gender e.t.c. 
+ * 
+ * <p>
+ * The criteria is used to evaluate if certain attributes of a patient meet a certain factor range. 
+ * For example, if criteria is of factor age(say age between 1-5), then the ranges only apply for this age group. 
+ * </p>
+ *
+ * @since 2.7.0
+ */
+@Entity
+@Table(name = "concept_reference_range")
+@Audited
+public class ConceptReferenceRange extends BaseReferenceRange implements Auditable, Retireable {
+	private static final long serialVersionUID = 47329L;
+
+	// Fields
+	@DocumentId
+	@Id
+	@Column(name = "concept_reference_range_id")
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Integer conceptReferenceRangeId;
+
+	@Column(name = "criteria")
+	private String criteria;
+
+	@ManyToOne(fetch = FetchType.LAZY)
+	@JoinColumn(name = "concept_id", nullable = false)
+	private Concept concept;
+
+	@ManyToOne
+	@JoinColumn(name = "retired_by")
+	private User retiredBy;
+
+	@Column(name = "date_retired")
+	private Date dateRetired;
```

Comment:
We are not going to retire data in this table. We shall only be retiring the concept. So this table should be left with only the fields in the concept_numeric table plus criteria, concept, and concept reference id.

Reply:
What about audit fields? e.g. creator, dateCreated, dateChanged?

Reply:
Since this is an extension of the concept, just like concept_numeric, i would not include those fields.